src/Control/Email/Email.php

```diff
@@ -31,738 +16,646 @@ class Email extends ViewableData
+    }
 
+    /**
+     * Used by {@link SSViewer} templates to detect if we're rendering an email template rather than a page template
+     *
+     * @return bool
+     */
+    public function IsEmail()
+    {
+        return true;
+    }
 
-        $to = $this->to;
-        $from = $this->from;
-        $subject = $this->subject;
-        if ($sendAllTo = $this->config()->send_all_emails_to) {
-            $subject .= " [addressed to $to";
-            $to = $sendAllTo;
-            if ($this->cc) {
-                $subject .= ", cc to $this->cc";
-            }
-            if ($this->bcc) {
-                $subject .= ", bcc to $this->bcc";
-            }
-            $subject .= ']';
-            unset($headers['Cc']);
-            unset($headers['Bcc']);
-        } else {
-            if ($this->cc) {
-                $headers['Cc'] = $this->cc;
-            }
-            if ($this->bcc) {
-                $headers['Bcc'] = $this->bcc;
-            }
+    /**
+     * @param string $template
+     * @param array|ViewableData $data
+     * @param callable|null $callback
+     * @return static
+     */
+    public static function create_from_callback($template, $data, $callback = null)
+    {
+        // create our email message from the arguments
+        $message = static::create();
+        $message->setTemplate($template);
+        $message->setData($data);
+
+        // if a callback is provided, call it against the message
+        if (is_callable($callback)) {
+            $callback($message);
         }
 
+        return $message;
+    }
 
-        if ($ccAllTo = $this->config()->cc_all_emails_to) {
-            if (!empty($headers['Cc']) && trim($headers['Cc'])) {
-                $headers['Cc'] .= ', ' . $ccAllTo;
-            } else {
-                $headers['Cc'] = $ccAllTo;
-            }
-        }
-
-        if ($bccAllTo = $this->config()->bcc_all_emails_to) {
-            if (!empty($headers['Bcc']) && trim($headers['Bcc'])) {
-                $headers['Bcc'] .= ', ' . $bccAllTo;
-            } else {
-                $headers['Bcc'] = $bccAllTo;
-            }
+    /**
+     * Send the message to the recipients
+     *
+     * @return array|bool true if successful or array of failed recipients
+     */
+    public function send()
+    {
+        $this->getSwiftMessage()->setContentType('text/html');
+        //create plain text part
+        $this->getSwiftMessage()->addPart(
+            Convert::xml2raw($this->getSwiftMessage()->getBody()),
+            'text/plain',
+            'utf-8'
+        );
+        if (!$this->getBody()) {
+            $this->render();
         }
-
-        if ($sendAllfrom = $this->config()->send_all_emails_from) {
-            if ($from) {
-                $subject .= " [from $from]";
-            }
-            $from = $sendAllfrom;
+        $numFailed = Mailer::send_message($this);
+        if ($numFailed) {
+            return Mailer::get_inst()->getFailedRecipients();
```

Comment:
Would change this to `Injector::inst()->get(Mailer::class)->getFailedRecipients()` if Mailer is to become an interface.